community/lucene-index/src/main/java/org/neo4j/index/lucene/legacy/IndexType.java

```diff
@@ -288,36 +303,60 @@ Query query( String keyOrNull, Object value, QueryContext contextOrNull )
 
     abstract void addToDocument( Document document, String key, Object value );
 
-    public static Fieldable instantiateField( String key, Object value, Index analyzed )
+    public static IndexableField instantiateField( String key, Object value, FieldType fieldType )
     {
-        Fieldable field;
+        IndexableField field;
         if ( value instanceof Number )
         {
             Number number = (Number) value;
-            NumericField numberField = new NumericField( key, Store.YES, true );
             if ( value instanceof Long )
             {
-                numberField.setLongValue( number.longValue() );
+                field = new LongField( key, number.longValue(), Store.YES );
             }
             else if ( value instanceof Float )
             {
-                numberField.setFloatValue( number.floatValue() );
+                field = new FloatField( key, number.floatValue(), Store.YES );
             }
             else if ( value instanceof Double )
             {
-                numberField.setDoubleValue( number.doubleValue() );
+                field = new DoubleField( key, number.doubleValue(), Store.YES );
             }
             else
             {
-                numberField.setIntValue( number.intValue() );
+                field = new IntField( key, number.intValue(), Store.YES );
             }
-            field = numberField;
         }
         else
         {
-            field = new Field( key, value.toString(), Store.YES, analyzed );
+            field = new Field( key, value.toString(), fieldType );
+        }
+        // TODO: ensure  omitNorms on fieldType?
```

Comment:
@tinwelint  currently people can create fields with omitNorms false and can have a functionality that depends on it, do no think that we can and to remove it. wdyt?